content/java-interview-questions-2/4-new-java-features.md

```diff
@@ -1,37 +1,96 @@
 ## Question
-# What is Spring?
+# Which version of Java have you worked on ?
+--
+# Answer
+- you can explain whatever java version you have used like java 8,9,10 or 11.
+- out of these java 8 and java 11 are the long term support versions and most of the questions will be on these 2 versions.
+---
+    
+# Can you explain new features of java 8 ?
+--
+# Answer
+- Default and static methods in interface
+- lambda and functional interfaces
+- Stream API
+- Collections improved
+- Date time API is improved
+- Java IO operations improved
+---
 
+# What is functional interface ?
 --
+# Answer
+ - Functional interfaces are those which has only one abstract method but can have multiple implemented methods. 
+ - It's also called as SAM(single abstract method). 
+ - Functional interfaces enable the use of lambda expressions. 
+ - We can use @FunctionalInterface annotation for these interfaces.
+---
 
-# What is the interviewer looking for?
-- One
-- Two
-- Three
+# Why the static or default methods in interface are introduced in Java 8 ?
+--
+# Answer
+ -  In the java old version there is a lot of use of interfaces, and to provide the backword compatibility for these interfaces, default methods are introduced. 
```

Comment:
```suggestion
 -  Allows addition of new methods to old interfaces 
 - This prevents old implementations from breaking because of compiler errors due to unimplemented methods
```